### ISSUE TITLE
Cant build in iOS ?

### ISSUE THREAD

**Issue by kkmf-16** [vegidio-flutter/appodeal#25]
[!] CocoaPods could not find compatible versions for pod "appodeal_flutter":
  In Podfile:
    appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`)

Specs satisfying the `appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`)` dependency were found, but they required a higher minimum deployment target.

**Comment by vegidio** [comment]
Can you please paste the whole stack trace?

**Comment by kkmf-16** [comment]
Last login: Fri Apr  2 16:46:39 on console
kksworld@iMac-KKs mf-flutter % flutter build ios
 
Building com.my********** for device (ios-release)...
Automatically signing iOS for device deployment using specified development team
in Xcode project: 9BJ3W6****
Running pod install...                                              4.6s
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)

    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `appodeal_flutter` from
    `.symlinks/plugins/appodeal_flutter/ios`
    -> Fetching podspec for `firebase_auth` from
    `.symlinks/plugins/firebase_auth/ios`
    -> Fetching podspec for `firebase_core` from
    `.symlinks/plugins/firebase_core/ios`
    -> Fetching podspec for `firebase_messaging` from
    `.symlinks/plugins/firebase_messaging/ios`
    -> Fetching podspec for `flutter_email_sender` from
    `.symlinks/plugins/flutter_email_sender/ios`
    -> Fetching podspec for `flutter_webview_plugin` from
    `.symlinks/plugins/flutter_webview_plugin/ios`
    -> Fetching podspec for `geolocator` from `.symlinks/plugins/geolocator/ios`
    -> Fetching podspec for `location_permissions` from
    `.symlinks/plugins/location_permissions/ios`
    -> Fetching podspec for `path_provider` from
    `.symlinks/plugins/path_provider/ios`
    -> Fetching podspec for `share` from `.symlinks/plugins/share/ios`
    -> Fetching podspec for `url_launcher` from
    `.symlinks/plugins/url_launcher/ios`
    -> Fetching podspec for `webview_flutter` from
    `.symlinks/plugins/webview_flutter/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local
      because checking is only perfomed in repo update
    [!] CocoaPods could not find compatible versions for pod "appodeal_flutter":
      In Podfile:
        appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`)

    Specs satisfying the `appodeal_flutter (from
    `.symlinks/plugins/appodeal_flutter/ios`)` dependency were found, but they
    required a higher minimum deployment target.

    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:328:in `raise_error_unless_state'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:310:in `block in unwind_for_conflict'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:308:in `tap'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:308:in `unwind_for_conflict'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:684:in `attempt_to_activate'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:254:in `process_topmost_state'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:182:in `resolve'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolver.rb:43:in `resolve'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/resolver.rb:94:in `resolve'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer/analyzer.rb:1074:in `block in
    resolve_dependencies'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer/analyzer.rb:1072:in `resolve_dependencies'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer/analyzer.rb:124:in `analyze'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer.rb:414:in `analyze'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer.rb:239:in `block in resolve_dependencies'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer.rb:238:in `resolve_dependencies'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/installer.rb:160:in `install!'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/command/install.rb:52:in `run'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/claide-1.0.3
    /lib/claide/command.rb:334:in `run'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/lib/cocoapods/command.rb:52:in `run'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/cocoapods-1.
    10.1/bin/pod:55:in `<top (required)>'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/bin/pod:23:in `load'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/bin/pod:23:in `<main>'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/bin/ruby_executable_hooks:24:in `eval'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/bin/ruby_executable_hooks:24:in
    `<main>'

Error running pod install
kksworld@iMac-KKs mf-flutter % 

**_Pubspec_**

name: my_*******
description: My ****** app developed in Flutter.
version: 1.0.3+4

environment:
  sdk: ">=2.0.0-dev.68.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.2
  navigate: ^1.0.1
  dio: ^3.0.8
  sembast: ^2.4.7+6
  path_provider: ^1.5.1
  flutter_spinkit: ^4.1.1+1
  modal_progress_hud: ^0.1.3
  event_bus: ^1.1.1
  geolocator: ^5.3.2
  dots_indicator: ^1.2.0
  expandable: ^3.0.0+1
  webview_flutter: ^1.0.7
  flutter_html: ^0.11.1
  url_launcher: ^5.4.2
  firebase_core: ^0.4.5
  #  firebase_analytics: ^5.0.15
  firebase_auth: ^0.16.1
  firebase_messaging: ^6.0.16
  flutter_email_sender: ^3.0.1
  flutter_webview_plugin: ^0.3.11
  #  firebase_admob: ^0.9.3+4
  #  flutter_native_admob: ^2.1.0+3
  share: ^0.6.5+4
  flutter_swiper: ^1.1.6
  appodeal_flutter: ^0.6.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_launcher_icons: ^0.7.4

flutter_icons:
  image_path: "assets/appLogo.png"
  android: true
  ios: true

flutter:
  uses-material-design: true
  assets:

**Comment by vegidio** [comment]
Judging by this message:

```
Specs satisfying the `appodeal_flutter (from
`.symlinks/plugins/appodeal_flutter/ios`)` dependency were found, but they
required a higher minimum deployment target.
```

It seems that you need to increase the deployment target on iOS, however it's a little premature to say that. Can you also run `pod install` in your project folder and send me the output?

Another thing that I noticed is that you are using some outdated versions of a few Flutter libs, like **firebase_analytics**, **url_launcher** and **dio**. I was using similar outdated versions of those libs in my personal project and for some reason the newest Cocoapods version of Appodeal didn't play well with them. But my problems disappeared when I updated them to their latest version.

Please try if you can update these Flutter packages and see if the problem disappears.

**Comment by kkmf-16** [comment]
Last login: Fri Apr  2 21:08:07 on ttys004
kksworld@iMac-KKs ios % pod install
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "appodeal_flutter":
  In Podfile:
    appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`)

Specs satisfying the `appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`)` dependency were found, but they required a higher minimum deployment target.
kksworld@iMac-KKs ios %

**Comment by kkmf-16** [comment]
I can build without Appodeal plugin and Appodeal related codes. When i use appodeal plugin it shows build error. Both build logs are below

**Without Appodeal Plugin and Appodeal related Code**

kksworld@iMac-KKs mf-flutter % flutter build ios
Building com.my******** for device (ios-release)...
Automatically signing iOS for device deployment using specified development team
in Xcode project: 9BJ3******
Running pod install...                                              3.3s
Running Xcode build...                                                  
                                                   
 └─Compiling, linking and signing...                             
  145.7s (!)                                       
Xcode build done.                                           330.1s
Built
/Users/kksworld/Desktop/office_work/mf-flutter/build/ios/iphoneos/runner.app.


**With Appodeal Plugin and Appodeal related Code**

kksworld@iMac-KKs mf-flutter % flutter build ios
Building com.my********* for device (ios-release)...
Automatically signing iOS for device deployment using specified development team
in Xcode project: 9BJ3******
Running pod install...                                              1.6s
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)

    Finding Podfile changes
      A appodeal_flutter
      - Flutter
      - firebase_auth
      - firebase_core
      - firebase_messaging
      - flutter_email_sender
      - flutter_webview_plugin
      - geolocator
      - location_permissions
      - path_provider
      - share
      - url_launcher
      - webview_flutter

    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `appodeal_flutter` from
    `.symlinks/plugins/appodeal_flutter/ios`
    -> Fetching podspec for `firebase_auth` from
    `.symlinks/plugins/firebase_auth/ios`
    -> Fetching podspec for `firebase_core` from
    `.symlinks/plugins/firebase_core/ios`
    -> Fetching podspec for `firebase_messaging` from
    `.symlinks/plugins/firebase_messaging/ios`
    -> Fetching podspec for `flutter_email_sender` from
    `.symlinks/plugins/flutter_email_sender/ios`
    -> Fetching podspec for `flutter_webview_plugin` from
    `.symlinks/plugins/flutter_webview_plugin/ios`
    -> Fetching podspec for `geolocator` from `.symlinks/plugins/geolocator/ios`
    -> Fetching podspec for `location_permissions` from
    `.symlinks/plugins/location_permissions/ios`
    -> Fetching podspec for `path_provider` from
    `.symlinks/plugins/path_provider/ios`
    -> Fetching podspec for `share` from `.symlinks/plugins/share/ios`
    -> Fetching podspec for `url_launcher` from
    `.symlinks/plugins/url_launcher/ios`
    -> Fetching podspec for `webview_flutter` from
    `.symlinks/plugins/webview_flutter/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local
      because checking is only perfomed in repo update
    [!] CocoaPods could not find compatible versions for pod "appodeal_flutter":
      In Podfile:
        appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`)

    Specs satisfying the `appodeal_flutter (from
    `.symlinks/plugins/appodeal_flutter/ios`)` dependency were found, but they
    required a higher minimum deployment target.

    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:328:in `raise_error_unless_state'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:310:in `block in unwind_for_conflict'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:308:in `tap'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:308:in `unwind_for_conflict'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:684:in `attempt_to_activate'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:254:in `process_topmost_state'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolution.rb:182:in `resolve'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/molinillo-0.
    6.6/lib/molinillo/resolver.rb:43:in `resolve'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/res
    olver.rb:94:in `resolve'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller/analyzer.rb:1074:in `block in resolve_dependencies'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/use
    r_interface.rb:64:in `section'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller/analyzer.rb:1072:in `resolve_dependencies'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller/analyzer.rb:124:in `analyze'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller.rb:414:in `analyze'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller.rb:239:in `block in resolve_dependencies'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/use
    r_interface.rb:64:in `section'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller.rb:238:in `resolve_dependencies'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/ins
    taller.rb:160:in `install!'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/com
    mand/install.rb:52:in `run'
    /Users/kksworld/.rvm/rubies/ruby-2.7.0/lib/ruby/gems/2.7.0/gems/claide-1.0.3
    /lib/claide/command.rb:334:in `run'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/lib/cocoapods/com
    mand.rb:52:in `run'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/gems/cocoapods-1.10.1/bin/pod:55:in
    `<top (required)>'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/bin/pod:23:in `load'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/bin/pod:23:in `<main>'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/bin/ruby_executable_hooks:24:in `eval'
    /Users/kksworld/.rvm/gems/ruby-2.7.0/bin/ruby_executable_hooks:24:in
    `<main>'

Error running pod install
kksworld@iMac-KKs mf-flutter %

**Comment by vegidio** [comment]
If all your plugins are updated, but you keep getting this error message:

```
Specs satisfying the appodeal_flutter (from .symlinks/plugins/appodeal_flutter/ios) dependency were found, but they required a higher minimum deployment target.
```

Then I would suggest to check what is your deployment target and try to increase. Another that you can try is to install an old version of the Appodeal package to see if you get different results. Try version versions `0.4.1` and `0.3.1`.

Also, please clone this repo and try to build the example app.

**Comment by nemoryoliver** [comment]
Mine has conflict with GMA SDK I guess

```
!] CocoaPods could not find compatible versions for pod "Google-Mobile-Ads-SDK":
      In Podfile:
        appodeal_flutter (from `.symlinks/plugins/appodeal_flutter/ios`) was resolved to 0.6.0, which depends on
          APDGoogleAdMobAdapter (= 2.9.1.1) was resolved to 2.9.1.1, which depends on
            Google-Mobile-Ads-SDK (~> 8.1.0)

        flutter_native_admob (from `.symlinks/plugins/flutter_native_admob/ios`) was resolved to 0.0.1, which depends on
          Google-Mobile-Ads-SDK

        google_mobile_ads (from `.symlinks/plugins/google_mobile_ads/ios`) was resolved to 0.0.1, which depends on
          Google-Mobile-Ads-SDK (= 7.69.0)
```

**Comment by vegidio** [comment]
@nemoryoliver Unfortunately there's not I can do here. This plugin is using the dependency **Google-Mobile-Ads-SDK (~> 8.1.0)** but it seems you are also using another plugin that depends on an older version of the same pod **Google-Mobile-Ads-SDK (= 7.69.0)**. And Cocoapods (the dependency management used by Flutter on iOS) doesn't support 2 versions of the same dependency on the same project.

You can either:

1. Check if google_mobile_ads has a newer version with a more up-to-date dependency on Google-Mobile-Ads-SDK
2. You can fork one of the plugins and change the dependency to match one other.
